apiserver/authentication/jwt/jwt.go

```diff
@@ -50,32 +65,21 @@ type TokenEntity struct {
 	User names.UserTag
 }
 
-// TokenParser parses a jwt token returning the token and
-// entity derived from the token subject.
-type TokenParser interface {
-	// Parse parses the supplied token string and returns both the constructed
-	// jwt and the entity found within the token.
-	Parse(ctx context.Context, tok string) (jwt.Token, authentication.Entity, error)
-}
-
-func NewAuthenticator(refreshURL string) *JWTAuthenticator {
-	httpClient := &http.Client{
-		Timeout: 30 * time.Second,
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse
-		},
+// Authenticate implements EntityAuthenticator
+func (j *JWTAuthenticator) Parse(ctx context.Context, tok string) (jwt.Token, TokenEntity, error) {
+	parser, ok := j.parserGetter.Get()
```

Comment:
I am not sold on this interface. The parser is available and can be returned. I think it is the parsers job to return a status that tells you if it is configured or not.